src/components/editor/Bindings/SchemaInference/Dialog.tsx

```diff
@@ -0,0 +1,415 @@
+                    <FormattedMessage id="workflows.collectionSelector.schemaInference.header" />
+                </Typography>
+            </DialogTitle>
+
+            <DialogContent>
+                <Typography sx={{ mb: 2 }}>
+                    <FormattedMessage id="workflows.collectionSelector.schemaInference.message" />
+                </Typography>
+
+                <Typography sx={{ mb: 4 }}>
+                    <FormattedMessage id="workflows.collectionSelector.schemaInference.message.schemaDiff" />
+                </Typography>
+
+                {schemaUpdateErrored ? (
+                    <AlertBox
+                        severity="error"
+                        short
+                        title={
+                            <Typography>
+                                <FormattedMessage id="workflows.collectionSelector.schemaInference.alert.generalError.header" />
+                            </Typography>
+                        }
+                    >
+                        <MessageWithLink messageID="workflows.collectionSelector.schemaInference.alert.patchService.message" />
+                    </AlertBox>
+                ) : null}
+
+                <Box sx={{ my: 3 }}>
+                    <Box
+                        sx={{
+                            p: 1,
+                            height: 54,
+                            backgroundColor:
+                                monacoEditorHeaderBackground[
+                                    theme.palette.mode
+                                ],
+                        }}
+                    >
+                        <Stack
+                            direction="row"
+                            sx={{
+                                height: '100%',
+                                justifyContent: 'space-between',
+                                alignItems: 'center',
+                            }}
+                        >
+                            <CustomWidthTooltip
+                                title={currentCollection}
+                                placement="bottom-start"
+                            >
+                                <Typography noWrap sx={{ mr: 2 }}>
+                                    {currentCollection}
+                                </Typography>
+                            </CustomWidthTooltip>
+
+                            {loading ? (
+                                <Box sx={{ px: 1, pt: 1 }}>
+                                    <CircularProgress size="1.5rem" />
+                                </Box>
+                            ) : null}
+                        </Stack>
+                    </Box>
+
+                    {inferredSchema ? (
+                        <DiffEditor
+                            height={`${height}px`}
+                            original={stringifyJSON(collectionData.spec)}
+                            modified={stringifyJSON(inferredSchema)}
```

Comment:
Could these calls end up being pretty heavy? I am wondering if we should throw these into memos? I'll leave it up to you.